lib/ceedling/preprocessinator_includes_handler.rb

```diff
@@ -87,6 +104,18 @@ def extract_shallow_includes(make_rule)
     sdependencies.map! {|hdr| hdr.gsub('\\','/') }
     list += sdependencies
 
+    if @configurator.project_config_hash.has_key?(:project_auto_link_deep_dependencies) && @configurator.project_config_hash[:project_auto_link_deep_dependencies]
+      # Find corresponding source files from removed header files (if they exist):
+      removed_headers.find_all do |removed_header|
+        source_file = removed_header.delete_suffix(hdr_ext) + src_ext
+        if File.exist?(source_file)
+          other_make_rule = self.form_shallow_dependencies_rule(source_file)
+          other_deps = self.extract_shallow_includes(other_make_rule, removed_headers + real_headers + mock_headers)
```

Comment:
```suggestion
          other_deps = self.extract_shallow_includes(other_make_rule, ignore_list + removed_headers + real_headers + mock_headers)
```

To make sure that ignore list grows with every new file inspected and to cut cycles (not sure if they are possible)